aider/io.py

```diff
@@ -360,3 +366,27 @@ def append_chat_history(self, text, linebreak=False, blockquote=False):
         if self.chat_history_file is not None:
             with self.chat_history_file.open("a", encoding=self.encoding) as f:
                 f.write(text)
+
+def read_block():
+    import sys
+
+    print("Use CTRL+D, CTRL+Z or CTRL+Z + ENTER to finish...")
+
+    echo_buffer = ""
```

Comment:
This simulates an echo on windows (with backspace support). Otherwise the user won't see what is being typed/pasted into the terminal window.
Terminal behaviour may be different depending on the OS / environment:
On Windows 10, it is required to use CTRL+Z + ENTER to generate the EOF.